src/NuGet.Clients/NuGet.VisualStudio.Implementation/PreinstalledRepositoryProvider.cs

```diff
@@ -71,7 +71,24 @@ public void AddFromRegistry(string keyName, bool isPreUnzipped)
             _repositories.Add(source);
         }
 
+        /// <summary>
+        /// Adds a NuGet package source from a Visual Studio Extension
+        /// </summary>
+        /// <param name="provider">Not Used</param>
+        /// <param name="extensionId">VSPackage Extension GUID</param>
+        /// <remarks>https://docs.microsoft.com/en-us/nuget/visual-studio-extensibility/visual-studio-templates</remarks>
+        [Obsolete("Use overload without provider argument instead")]
         public void AddFromExtension(ISourceRepositoryProvider provider, string extensionId)
```

Comment:
Is this an implementation of an API?

NuGet.VisualStudio.Implementation doesn't ship publicly.

Reply:
Removed.